assignments/prototypes.js

```diff
@@ -8,40 +8,46 @@
   Each constructor function has unique properites and methods that are defined in their block comments below:
 */
   
-/*
-  === GameObject ===
-  * createdAt
-  * dimensions
-  * destroy() // prototype method -> returns the string 'Object was removed from the game.'
-*/
 
-/*
-  === CharacterStats ===
-  * hp
-  * name
-  * takeDamage() // prototype method -> returns the string '<object name> took damage.'
-  * should inherit destroy() from GameObject's prototype
-*/
+  function GameObject (attributes) {
+    this.createdAt = attributes.createdAt;
+    this.dimesions = attribuetes.dimensions;
+  }
+  
+ 
+  GameObject.prototype.destroy = function() {
+    return 'Object was removed from the game';
+}
 
-/*
-  === Humanoid ===
-  * faction
-  * weapons
-  * language
-  * greet() // prototype method -> returns the string '<object name> offers a greeting in <object language>.'
-  * should inherit destroy() from GameObject through CharacterStats
-  * should inherit takeDamage() from CharacterStats
-*/
+  
+  function CharacterStats (characterStatsAttributes) {
+    this.hp = characterStatsAttributes.hp;
+    this.name = characterStatsAttributes.name;
+    GameObject.call(this, attributes)
+  }
+  
+  CharacterStats.prototype = Object.create(GameObject.prototype);
  
-/*
-  * Inheritance chain: Humanoid -> CharacterStats -> GameObject
-  * Instances of Humanoid should have all of the same properties as CharacterStats and GameObject.
-  * Instances of CharacterStats should have all of the same properties as GameObject.
-*/
+  CharacterStats.prototype.takeDamage = function () {
+    return `${this.name} tooke damage.`;
+  }
+    
+  function Humanoid (humanoidAttributes) {
+    this.faction = humanoidAttributes.faction;
+    this.weapons = humanoidAttributes.weapons;
+    this.language = humanoidAttributes.language;
+    CharacterStats.call(this, characterStatsAttributes)
+  }
 
-//Test you work by uncommenting these 3 objects and the list of console logs below:
+  Humanoid.prototype = Object.create(CharacterStats.prototype);
+
+  Humanoid.prototype.greet = function () {
+    returns `${Humanoidthis.name} offers a greeting in ${this.language}`;
```

Comment:
This is correct, but you'd have to take out the `Humanoid` on line 45! Also, typo: `returns` should be `return` 🙂